index.js

```diff
@@ -248,6 +282,36 @@ function $asBooleanNullable (bool) {
   return bool === null ? null : $asBoolean(bool)
 }
 
+function $asDatetime (date) {
+  if (date instanceof Date) {
+    return '"' + date.toISOString() + '"'
+  } else if (date instanceof moment) {
+    return '"' + date.toISOString() + '"'
+  } else {
+    return $asString(date)
+  }
+}
+
+function $asDate (date) {
+  if (date instanceof Date) {
+    return '"' + moment(date).format('YYYY-MM-DD') + '"'
+  } else if (date instanceof moment) {
+    return '"' + date.format('YYYY-MM-DD') + '"'
+  } else {
+    return $asString(date)
+  }
+}
+
+function $asTime (date) {
+  if (date instanceof Date) {
+    return '"' + moment(date).format('HH:mm:ss') + '"'
+  } else if (date instanceof moment) {
```

Comment:
Unfortunately this check will not work because a user can have more than one version of moment in their dep tree.